tests/test_tsp.py

```diff
@@ -136,6 +136,41 @@ def test_k3(self):
 
         self.assertEqual(ground_count, len(min_routes))
 
+    def test_k3_bidirectional(self):
```

Comment:
So this tests the original interface, which is fine. But we should also add tests for `nx.Graph` with missing edges and `nx.MultiGraph` both with and without missing edges. Also test whether my user-specified `missing_edge_weight` gets used.